### DISCUSSION TITLE
deploying to AWS lightsail

### DISCUSSION THREAD

**Discussion by dawid-dol** [pocketbase/pocketbase#1865] [Q&A]
Hi Folk,

I'm trying to deploy my pocketbase database to aws lightsail. Just looking for an easy way to do it really, don't want to have to tinker with EC2 and all that. 

I run 
./pocketbase serve --https=<My-public-ip>:8080 --debug
I got it all on it but I keep getting this error 

listen tcp<my-public-ip>:5050: bind: cannot assign requested address 

Not really sure how to fix it or what to do here?

any help is welcome, thanks!

Edit, 

I also gave app runner a go and it works great! however, if I push an update on my container image with a new version for example, my data will be delete, no? since the app runner instance will have the data from the current isntace running?

**Comment by ganigeorgiev** [comment]
Unfortunately, I'm not familiar with AWS Lightsail and its deployment workflow and I'm not sure how to help.

Just for the test, you can try specifing the public IP of your vm when starting the executable or assign `0.0.0.0:8080` to explicitly bind to all IPv4 interfaces.

The error could be also caused by a firewall setting or some other configuration that may need to change in the Lightsail console.


> I also gave app runner a go and it works great! however, if I push an update on my container image with a new version for example, my data will be delete, no? 

As far as I know, AWS App Runner has ephemeral storage, meaning that you may have to mount an additional persistent volume, eg. EFS (_but I have very limited experience with App Runner and whether it is possible at all_). In general, your application data lives in the `pb_data` directory and as long as this directory is persisted your application will continue from the last stored state on new version push.

**Reply by dawid-dol** [reply]
Hello, thank you for your response.

yeah so I ran it as 0.0.0.0 and that worked :) However, it only works for HTTP. For https I get 'Client sent an HTTP request to an HTTPS server.'

I am assuming this is an error on my side but any tips are welcome 😄 

Thank you again.

**Reply by ganigeorgiev** [reply]
Have you configured a SSL/TLS certificate in the AWS Lightsail console (ex. https://lightsail.aws.amazon.com/ls/docs/en_us/articles/understanding-tls-ssl-certificates-in-lightsail-https)?

Note that PocketBase also comes with auto Let's encrypt certificate generation but you need to start the executable with `--https` flag and to have a valid public domain name binded to your vm (_check for example the ["Minimal setup guide"](https://pocketbase.io/docs/going-to-production/#minimal-setup)_), but I'm not sure if that will work with AWS Lightsail.

**Reply by dawid-dol** [reply]
yeah, the --https doesn't seem to work with lightsail I just get "The connection for this site is not secure " error so it seems AWS doesnt like working with  Let's encrypt 

on the server side I get the  TLS handshake error from <public IP>  acme/autocert: missing server name

**Reply by ganigeorgiev** [reply]
Note that when using `--https` you need to provide a valid public domain name (aka. you can't bind with the IP).
I'm not sure what is the procedure with Lightsail (if you haven't already, you can try checking the above Lightwail docs url or some of its related resources, eg. https://lightsail.aws.amazon.com/ls/docs/en_us/articles/create-tls-ssl-certificate-and-attach-to-lightsail-load-balancer-https).

**Reply by dawid-dol** [reply]
yeah, ill need to do that thank you!


on another note, while I am here. in Filestorage i can connect an s3 bucket, is there an integrated way to store collections data in s3? just a backup or should I manually set that up?

**Reply by ganigeorgiev** [reply]
> is there an integrated way to store collections data in s3?

Out of the box, only the uploaded files can be stored on s3.
The SQLite database (`pb_data/data.db`) **can't** be accessed/mounted from external filestorage (aka. the database needs to be on the "disk").
If you just want to backup your app data to s3, then you can run periodically (eg. via cron) `rclone` or other similar cli tool to backup your `pb_data` directory to your bucket, but there is no integrated PocketBase command for that.

**Reply by boustanihani** [reply]
> AWS App Runner

Could you please share the setup tp make pocketbase work on AWS App Runner?

**Comment by boustanihani** [comment]
A relevant discussion also worth checking:

Host Pocketbase on AWS using Lambda Web Adapter and Elastic File System (EFS) => https://github.com/pocketbase/pocketbase/discussions/7246

**Reply by ganigeorgiev** [reply]
@boustanihani Please don't spam like that!
